src/levanter/config.py

```diff
@@ -93,26 +93,51 @@ def _maybe_get_config_path_and_cmdline_args(args: List[str]):
     from the cmdline args so that draccus doesn't try to load it as a config path and return it separately here
     along with the modified cmdline args.
+    We also accept ... --configs <config1> <config2> ... and concatenate them into a single config.
     """
-    if "--config_path" not in args and "--config" not in args:
+    if "--config_path" not in args and "--config" not in args and "--configs" not in args:
         return None, args
     else:
-        try:
-            config_path_index = args.index("--config_path")
-        except ValueError:
-            config_path_index = args.index("--config")
-
-        config_path = args[config_path_index + 1]
-
-        if urllib.parse.urlparse(config_path).scheme:
-            fs: AbstractFileSystem
-            fs, fs_path = fsspec.core.url_to_fs(config_path)
-            temp_file = tempfile.NamedTemporaryFile(prefix="config", suffix=".yaml", delete=False)
-            atexit.register(lambda: os.unlink(temp_file.name))
-            fs.get(fs_path, temp_file.name)
-            config_path = temp_file.name
+        config_args = ["--config_path", "--config", "--configs"]
+        found_indices = [args.index(arg) for arg in config_args if arg in args]
+        if len(found_indices) > 1:
+            raise ValueError(f"Multiple config args found in {args}")
+        config_path_index = found_indices[0] + 1
 
+        config_paths = []
         args = args.copy()
-        del args[config_path_index]
-        del args[config_path_index]
-        return config_path, args
+        del args[config_path_index - 1]
+        config_path_index -= 1
+
+        while config_path_index < len(args) and not args[config_path_index].startswith("-"):
+
+            config_path = args[config_path_index]
+
+            if urllib.parse.urlparse(config_path).scheme:
+                fs: AbstractFileSystem
+                fs, fs_path = fsspec.core.url_to_fs(config_path)
+                temp_file = tempfile.NamedTemporaryFile(prefix="config", suffix=".yaml", delete=False)
+                atexit.register(lambda: os.unlink(temp_file.name))
+                fs.get(fs_path, temp_file.name)
+                config_path = temp_file.name
+
+            config_paths.append(config_path)
+            del args[config_path_index]
+
+        merged_config_path = None
+
+        if len(config_paths) == 1:
+            merged_config_path = config_paths[0]
+        elif len(config_paths) > 1:
+            # merge the configs by concatenating them
+            temp_merged_config_path = tempfile.NamedTemporaryFile(prefix="config_merged", suffix=".yaml", delete=False)
+            atexit.register(lambda: os.unlink(temp_merged_config_path.name))
+            with open(temp_merged_config_path.name, "w") as f:
+                for config_path in config_paths:
+                    with open(config_path) as config_file:
+                        f.write(config_file.read())
```

Comment:
I'm assuming you are literally stitching multiple config files together?

e.g. we must have
```yaml
#data.yaml
data:
    cache_dir: ...
    ...
```

```yaml
#model.yaml
model:
    hidden_dim: 2048
    ...
```

Is the purpose of this PR for stitching top level configs (e.g. `model`, `data`, `trainer`) or do we also want to support stitching sub-configs (e.g. `data.train_weights.yaml`, `data.configs.yaml` ...)? I think it's better to utilize the yaml `!include` syntax, or else users need to pay extra attention on the order of the config they put in...

Reply:
agree with jason

Reply:
I think the problem with using `!include` is that when we want to do scaling laws, we would have to create a large number of configs. 

I think the vision I had was:

1. You can run with config just as normal run (Use `!include` and stuff)
2. We give multiple configs and stitch them together to make one config file.

Reply:
it's a good point